rust/worker/src/compactor/scheduler.rs

```diff
@@ -0,0 +1,397 @@
+use crate::compactor::task::Task;
+use crate::log::log::CollectionInfo;
+use crate::log::log::CollectionRecord;
+use crate::log::log::Log;
+use crate::sysdb::sysdb::SysDb;
+use crate::system::Component;
+use crate::system::ComponentContext;
+use crate::system::Handler;
+use async_trait::async_trait;
+use parking_lot::Mutex;
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use std::time::Duration;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub(crate) struct Scheduler {
+    log: Box<dyn Log>,
+    sysdb: Box<dyn SysDb>,
+    policy: SchedulerPolicy,
+    task_queue: Arc<Mutex<Vec<Task>>>,
+    max_queue_size: usize,
+    schedule_interval: Duration,
+}
+
+impl Scheduler {
+    pub(crate) fn new(
+        log: Box<dyn Log>,
+        sysdb: Box<dyn SysDb>,
+        policy: SchedulerPolicy,
+        max_queue_size: usize,
+        schedule_interval: Duration,
+    ) -> Scheduler {
+        Scheduler {
+            log,
+            sysdb,
+            policy,
+            task_queue: Arc::new(Mutex::new(Vec::with_capacity(max_queue_size))),
+            max_queue_size,
+            schedule_interval,
+        }
+    }
+
+    async fn get_collections_with_new_data(&mut self) -> Vec<CollectionInfo> {
+        let collections = self.log.get_collections_with_new_data().await;
+        // TODO: filter collecitons based on memberlist
+        let collections = match collections {
+            Ok(collections) => collections,
+            Err(e) => {
+                // TODO: Log error
+                println!("Error: {:?}", e);
+                return Vec::new();
+            }
+        };
+        collections
+    }
+
+    async fn verify_and_enrich_collections(
```

Comment:
nit: can we add some /// doc on this - so what "verify and enrich means is clear"